students/ShinTran/lesson04/kata.py

```diff
@@ -0,0 +1,78 @@
+'''
+Shin Tran
+Python 210
+Lesson 4 Assignment
+'''
+
+
+import string # For punctuation
+import random # For a random number generator
+
+# To store all the two word combo as keys and the following word as a value
+word_dict = {}
+
+
+# Opens a text file, parses though it line by line, storing two consecutive
+# words as keys, and the following word as a value
+def open_file():
+	global word_dict
+	sentence = []
+	contents = open('sherlock.txt')
+	#contents = open('sherlock_small.txt')
+	for line in contents:
+		unmod_sentence = line
+		for c in string.punctuation:
+			unmod_sentence = unmod_sentence.replace(c,' ')
+		sentence = unmod_sentence.split()
+		for i in range(0, len(sentence)-2):
+			the_key = sentence[i] + " " + sentence[i+1]
+			the_value = sentence[i+2]
+			if the_key not in word_dict:
+				word_dict.setdefault(the_key,[]).append(the_value)
+		sentence = []
+	contents.close()
+#	for key, value in word_dict.items():
+#		print(key + ": " + str(value))
+
+
+# Picks a key by random initially, appends those two words to a list,
+# then pulls the value and appends that to a list as well, take the 2nd
+# word of the key and the value, makes that the new key and repeats
+# based on the word count that's randomly generated, max of 15 words,
+# the loop breaks if no key/value is found
+def make_sentence():
+	global word_dict
+	the_sentence = []
+	first_set = random.choice(list(word_dict.keys()))
+	third_word = random.choice(word_dict[first_set])
+	the_sentence.append(first_set.split()[0])
+	the_sentence.append(first_set.split()[1])
+	the_sentence.append(third_word)
+	word_count = random.randint(5,14)
+	temp_key = ""
+	for i in range(1, word_count):
+		temp_key = str(the_sentence[i]) + " " + str(the_sentence[i+1])
+		next_word = word_dict.get(temp_key, 'NoKeyFound')
+		if next_word != 'NoKeyFound':
+			the_sentence.append(next_word[0])
+		else:
+			break
```

Comment:
please ensure your editor is set to 4 spaces (no tabs); all of the code is over-indented.

Reply:
I have no idea what happened there, it's always set to 4 spaces by default from the first or second week.